parachain/pallets/omni-account/src/lib.rs

```diff
@@ -396,6 +482,55 @@ pub mod pallet {
 
 			Ok(member_accounts)
 		}
+
+		fn ensure_permission(
+			call: &<T as Config>::RuntimeCall,
+			member_account_hash: H256,
+		) -> Result<(), Error<T>> {
+			let member_permissions = MemberAccountPermissions::<T>::get(member_account_hash);
+
+			ensure!(
+				member_permissions.iter().any(|permission| permission.filter(call)),
+				Error::<T>::NoPermission
+			);
+
+			match call.is_sub_type() {
+				Some(Call::add_account { permissions: ref new_account_permissions, .. }) => {
+					// If member has default permission, they can add accounts with any permission
+					if member_permissions.contains(&T::Permission::default()) {
+						return Ok(());
+					}
+					match new_account_permissions {
+						Some(new_permissions) => {
+							// an account can only add another account with the same or less permissions
```

Comment:
Do you consider using `is_superset()`?

Reply:
I did but since `is_superset` expects a single entry and we need to compare it with multiple entries, we'll still have to keep part of the logic here to make sure we check all the permissions. (I wanted to support multiple permissions instead of just one to make it more flexible and easier to extend)